packages/server/modules/emails/repositories/index.ts

```diff
@@ -12,24 +14,25 @@ export type EmailVerificationRecord = {
 /**
  * Attempt to find a valid & pending token entry for email verification
  */
-export async function getPendingToken(params: { token?: string; email?: string }) {
-  const { token, email } = params
-  if (!token && !email) throw new InvalidArgumentError('Token & email is empty')
+export const getPendingTokenFactory =
+  (deps: { db: Knex }): GetPendingToken =>
+  async ({ token, email }) => {
+    if (!token && !email) throw new InvalidArgumentError('Token & email is empty')
 
-  const aWeekAgo = dayjs().subtract(1, 'week')
+    const aWeekAgo = dayjs().subtract(1, 'week')
 
-  const q = EmailVerifications.knex<EmailVerificationRecord>()
-    .where(EmailVerifications.col.createdAt, '>', aWeekAgo.toISOString())
-    .first()
+    const q = EmailVerifications.knex<EmailVerificationRecord>(deps.db)
```

Comment:
here we should use the injected db
I know that technically behind the scene it is using it
But the idea is to completely remove things like `EmailVerification.knex` from the codebase
We should always use `db(tableName)`

Reply:
@alemagio It is using the injected DB, so why remove the useful helper?